hnn_core/viz.py

```diff
@@ -887,18 +892,26 @@ def plot_cell_morphology(cell, ax, color=None, show=True):
     if isinstance(color, dict):
         section_colors = color
 
+    _validate_type(pos, tuple, 'pos')
+    if isinstance(pos, tuple):
+        if len(pos) != 3:
+            raise ValueError('pos must be a tuple of 3 elements')
+        for pos_idx in pos:
+            _validate_type(pos_idx, (float, int), 'pos[idx]')
+
     # Cell is in XZ plane
-    ax.set_xlim((cell.pos[1] - 250, cell.pos[1] + 150))
-    ax.set_zlim((cell.pos[2] - 100, cell.pos[2] + 1200))
+    ax.set_xlim((pos[0] + xlim[0], pos[0] + xlim[1]))
+    ax.set_zlim((pos[1] + zlim[0], pos[1] + zlim[1]))
+    ax.set_ylim((pos[2] + ylim[0], pos[2] + ylim[1]))
 
     for sec_name, section in cell.sections.items():
         linewidth = _linewidth_from_data_units(ax, section.diam)
         end_pts = section.end_pts
         xs, ys, zs = list(), list(), list()
         for pt in end_pts:
-            dx = cell.pos[0] - cell.sections['soma'].end_pts[0][0]
-            dy = cell.pos[1] - cell.sections['soma'].end_pts[0][1]
-            dz = cell.pos[2] - cell.sections['soma'].end_pts[0][2]
+            dx = pos[0] - cell.sections['soma'].end_pts[0][0]
+            dy = pos[1] - cell.sections['soma'].end_pts[0][1]
+            dz = pos[2] - cell.sections['soma'].end_pts[0][2]
```

Comment:
It doesn't seem like these lines need to be nested in this for loop? Seems like all of those values are static. Should they be using pt instead of pos? Or the iterated section?